xradar/io/backends/iris.py

```diff
@@ -3917,6 +3922,72 @@ def _get_iris_group_names(filename):
     return keys
 
 
+def _get_required_root_dataset(ls_ds, optional=True):
+    """Extract Root Dataset."""
+    # keep only defined mandatory and defined optional variables per default
+    data_var = set(
+        [x for xs in [sweep.variables.keys() for sweep in ls_ds] for x in xs]
+    )
+    remove_root = set(data_var) ^ set(required_root_vars)
+    if optional:
+        remove_root ^= set(optional_root_vars)
+    remove_root ^= {"sweep_number", "fixed_angle"}
+    remove_root &= data_var
+    root = [sweep.drop_vars(remove_root) for sweep in ls_ds]
+    root_vars = set(
+        [x for xs in [sweep.variables.keys() for sweep in root] for x in xs]
+    )
+    # rename variables
+    # todo: find a more easy method not iterating over all variables
+    for k in root_vars:
+        rename = optional_root_vars.get(k, None)
+        if rename:
+            root = [sweep.rename_vars({k: rename}) for sweep in root]
+
+    root_vars = set(
+        [x for xs in [sweep.variables.keys() for sweep in root] for x in xs]
+    )
+    ds_vars = [sweep[root_vars] for sweep in ls_ds]
+
+    root = xr.concat(ds_vars, dim="sweep")
+    # keep only defined mandatory and defined optional attributes per default
+    attrs = root.attrs.keys()
+    remove_attrs = set(attrs) ^ set(required_global_attrs)
+    if optional:
+        remove_attrs ^= set(optional_root_attrs)
+    for k in remove_attrs:
+        root.attrs.pop(k, None)
+
+    # handle sweep variables and dimension
+    root = root.rename_vars({"sweep_number": "sweep_group_name"})
```

Comment:
@aladinor Please see further discussion in #164. We should remove `sweep_group_name` and `sweep_fixed_angle` from xradar data model.

Reply:
Should I remove it from model.py file ?
https://github.com/openradar/xradar/blob/0145954f3847abd89208c46055241aeae51316c8/xradar/model.py#L112C1-L121C2

Reply:
Yes, but this might show usages in other parts of xradar. If you need any help cleaning up the mess, let us know.